File: processlife/src/main/java/com/baichang/android/process/MainActivity.java
package com.baichang.android.process;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import com.baichang.android.process.service.MainService;

public class MainActivity extends AppCompatActivity {

  private static final String TAG = MainActivity.class.getSimpleName();

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_main);
    startService(new Intent(this, MainService.class));
  }

  @Override
  protected void onDestroy() {
    super.onDestroy();
    Log.d(TAG, "被杀了....");
    Log.d(TAG, "抢救中....");
    startService(new Intent(this, MainService.class));
  }
}

File: kotlin/src/main/java/com/example/kotlin/MainActivity.kt
package com.example.kotlin

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import java.io.File

class MainActivity : AppCompatActivity() {

  var quantity = 5
  val price: Double = 20.3
  val name: String = "大米"

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.activity_main)
//    println("Hello Kotlin")
//    Person("iCong").printName()
//
//    println("单价:$price")
//    println("数量:$quantity")
//    println("产品：$name 总价:${quantity * price}")
    read()
  }

  fun read() {
    File("build.gradle")
        .readText()
        .filterNot(Char::isWhitespace)
        .groupBy { it }
        .map {
          it.key to it.value.size
        }
        .forEach(::println)
  }
}
